projects/pastanaga-angular/src/lib/translate/translate.utils.ts

```diff
@@ -1,18 +1,20 @@
-import { TranslationEntries } from './translate.model';
+import { PA_TRANSLATIONS, Translation } from './translate.model';
 
-function deepMerge(from: TranslationEntries, to: TranslationEntries): TranslationEntries {
+function deepMerge(from: any, to: any): any {
     return Object.entries(from).reduce((all, [key, translation]) => {
         if (!all[key]) {
             all[key] = translation;
         } else if (typeof all[key] !== 'string' && typeof translation !== 'string') {
-            all[key] = deepMerge(translation, all[key] as TranslationEntries);
+            all[key] = deepMerge(translation, all[key]);
         }
         return all;
     }, to);
 }
 
-export const mergeTranslations = (translations: TranslationEntries[]): TranslationEntries => {
-    return translations.reduce((all, current) => {
-        return deepMerge(all, current);
-    }, {} as TranslationEntries);
+export const mergeTranslations = (translations: Translation[]) => {
+    translations.forEach((current) => {
+        Object.entries(current).forEach(
+            ([lang, entries]) => (PA_TRANSLATIONS[lang] = deepMerge(entries, PA_TRANSLATIONS[lang] || {})),
```

Comment:
I think this is the why for the "empty" `PA_TRANSLATIONS`.

I think that instead of "global" you should still do a reduce and update the module to be:

```ts
export class PaTranslateModule {
    static addTranslations(translations: Translation[]): ModuleWithProviders<PaTranslateModule> {
        const translations = mergeTranslations(translations);
        return {
            ngModule: PaTranslateModule,
            providers: [{
                provide: PA_TRANSLATIONS, useValue: translations
            }],
        };
    }
}
```

If it's the case that you have to work with a "shared" value from outside of the module, define empty `PA_TRANSLATIONS` in the module (not exported) and update `mergeTranslations` to accept a current value the additional translations should be merged into and still provide that value.